src/main/java/org/folio/rest/impl/InitAPIs.java

```diff
@@ -35,15 +51,43 @@ public void init(Vertx vertx, Context context, Handler<AsyncResult<Boolean>> res
         DatabindCodec.mapper().setConfig(deserializationConfig);
         DatabindCodec.prettyMapper().setConfig(deserializationConfig);
         SpringContextUtil.init(vertx, context, ApplicationConfig.class);
-        handler.complete();
+        SpringContextUtil.autowireDependencies(this, context);
+        deployConsumersVerticles(vertx).onSuccess(car -> {
+            handler.handle(Future.succeededFuture());
+            LOGGER.info("Consumer Verticles were successfully started");
+          })
+          .onFailure(th -> {
+            handler.handle(Future.failedFuture(th));
+            LOGGER.error("Consumer Verticles were not started", th);
+          });
       },
       result -> {
         if (result.succeeded()) {
           resultHandler.handle(Future.succeededFuture(true));
         } else {
-          logger.error("Failure to init API", result.cause());
+          LOGGER.error("Failure to init API", result.cause());
           resultHandler.handle(Future.failedFuture(result.cause()));
         }
       });
   }
+
+  private Future<?> deployConsumersVerticles(Vertx vertx) {
+    AbstractApplicationContext springContext = vertx.getOrCreateContext().get(SPRING_CONTEXT_KEY);
+    VerticleFactory verticleFactory = springContext.getBean(SpringVerticleFactory.class);
+    vertx.registerVerticleFactory(verticleFactory);
+
+    Promise<String> deployRawMarcChunkConsumer = Promise.promise();
+
+    vertx.deployVerticle(getVerticleName(verticleFactory, OrderConsumerVerticle.class),
+      new DeploymentOptions()
+        .setWorker(true)
+        .setInstances(orderConsumerInstancesNumber), deployRawMarcChunkConsumer);
+
+    return GenericCompositeFuture.all(Arrays.asList(
```

Comment:
Could we return just simple future instead of composite?

Reply:
completed